elements/lisk-api-client/src/codec.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import { codec, Schema } from '@liskhq/lisk-codec';
+import { hash } from '@liskhq/lisk-cryptography';
+import { RegisteredSchemas } from './types';
+
+export const getTransactionAssetSchema = (
+	transaction: Record<string, unknown>,
+	registeredSchema: RegisteredSchemas,
+): Schema => {
+	const txAssetSchema = registeredSchema.transactionsAssets.find(
+		assetSchema =>
+			assetSchema.moduleID === transaction.moduleID && assetSchema.assetID === transaction.assetID,
+	);
+	if (!txAssetSchema) {
+		throw new Error(
+			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+			`ModuleID: ${transaction.moduleID} AssetID: ${transaction.assetID} is not registered.`,
+		);
+	}
+	return txAssetSchema.schema;
+};
+
+export const decodeTransaction = (
+	encodedTransaction: Buffer,
+	registeredSchema: RegisteredSchemas,
```

Comment:
as this seems to hold several schemas it would probably be better for it to be plural `registeredSchemas`